homeassistant/components/aladdin_connect/cover.py

```diff
@@ -79,27 +80,88 @@ class AladdinDevice(CoverEntity):
     _attr_device_class = CoverDeviceClass.GARAGE
     _attr_supported_features = SUPPORTED_FEATURES
 
-    def __init__(self, acc: AladdinConnectClient, device: DoorDevice) -> None:
+    def __init__(
+        self, acc: AladdinConnectClient, device: DoorDevice, entry: ConfigEntry
+    ) -> None:
         """Initialize the Aladdin Connect cover."""
         self._acc = acc
+
         self._device_id = device["device_id"]
         self._number = device["door_number"]
         self._attr_name = device["name"]
         self._attr_unique_id = f"{self._device_id}-{self._number}"
 
-    def close_cover(self, **kwargs: Any) -> None:
+    async def async_added_to_hass(self) -> None:
+        """Connect Aladdin Connect to the cloud."""
+
+        @callback
+        async def update_callback() -> None:
+            """Schedule a state update."""
+            self.async_write_ha_state()
+
+        self._acc.register_callback(update_callback, self._number)
+        await self._acc.get_doors(self._number)
+        self.async_write_ha_state()
+
+    async def async_will_remove_from_hass(self) -> None:
+        """Close Aladdin Connect before removing."""
+        await self._acc.close()
+
+    async def async_close_cover(self, **kwargs: Any) -> None:
         """Issue close command to cover."""
-        self._acc.close_door(self._device_id, self._number)
+        await self._acc.close_door(self._device_id, self._number)
 
-    def open_cover(self, **kwargs: Any) -> None:
+    async def async_open_cover(self, **kwargs: Any) -> None:
         """Issue open command to cover."""
-        self._acc.open_door(self._device_id, self._number)
+        await self._acc.open_door(self._device_id, self._number)
 
-    def update(self) -> None:
+    async def async_update(self) -> None:
         """Update status of cover."""
-        status = STATES_MAP.get(
-            self._acc.get_door_status(self._device_id, self._number)
+        await self._acc.get_doors(self._number)
+
+    @property
+    def extra_state_attributes(self) -> Mapping[str, Any] | None:
+        """Update state attributes."""
+        extra_attributes = {}
+        extra_attributes.update(
+            {
+                "battery_level": self._acc.get_battery_status(
```

Comment:
Batter level and RSSI should be separate sensor entities. Please add them in a separate PR.

Reply:
will do.